dataprofiler/tests/profilers/test_categorical_column_profile.py

```diff
@@ -375,6 +373,32 @@ def test_unalikeability(self):
         profile.update(df_categorical)
         self.assertEqual(profile.unalikeability, 2*(10 + 15 + 6)/90)
 
+    def test_top_k_categories_change(self):
+        options = CategoricalOptions()
+        df_series = pd.Series(
+            ["a", "a", "b", "c", "d", "e", "e", "e", "f", "g"])
+        profile = CategoricalColumn(df_series.name, options)
+        profile.update(df_series)
+        self.assertEqual(
+            len(profile.profile['statistics']['categorical_count']), 7)
+        profile._top_k_categories = 6
+        self.assertEqual(
+            len(profile.profile['statistics']['categorical_count']), 6)
+
+        options = CategoricalOptions(top_k_categories=6)
```

Comment:
I would change these to:
`options = CategoricalOptions()` just once at the top, and then for each case you can just do
`options.top_k_categories = ___` so that you only create 1 options object and can keep reusing it. This also won't have to change if we decide to take away the parameter from the option init.